### ISSUE TITLE
console warning message when using ComboBox 

### ISSUE THREAD

**Issue by konghu** [IBM/carbon-components-react#1804]
## Detailed description

> Describe in detail the issue you're having.

> Is this issue related to a specific component?

> What did you expect to happen? What happened instead? What would you like to see changed?

> What browser are you working in?

> What version of the Carbon Design System are you using?

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
> https://codesandbox.io/s/github/IBM/carbon-components-react/tree/master/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by MichaelRowden** [comment]
@emyarod I've created a sandbox at https://codesandbox.io/s/ppnv5pon7
I too am experiencing this JS console warning and have included the reproduction steps below.

## Reproduction Steps:

1. Go to a page with the `ComboBox` component such as https://codesandbox.io/s/ppnv5pon7 (do not use http://react.carbondesignsystem.com as the warning does not appear there)
2. Open the JS console
3. Open the `ComboBox` component
4. Click off the `ComboBox` component without making a selection
5. The JS console will have the warning of ` Warning: An update (setState, replaceState, or forceUpdate) was scheduled from inside an update function. Update functions should be pure, with zero side-effects. Consider using componentDidUpdate or a callback.` (full warning message available below)

The warning seems to be generated around this code inside the `handleOnInputValueChange` function: https://github.com/IBM/carbon-components-react/blob/master/src/components/ComboBox/ComboBox.js#L189

## Full Warning Message:
```
proxyConsole.js:72 Warning: An update (setState, replaceState, or forceUpdate) was scheduled from inside an update function. Update functions should be pure, with zero side-effects. Consider using componentDidUpdate or a callback.
function.console.(anonymous function)	@	proxyConsole.js:72
r.(anonymous function)	@	index.js:26
warningWithoutStack	@	react-dom.development.js:506
enqueueUpdate	@	react-dom.development.js:15993
enqueueSetState	@	react-dom.development.js:11140
Component.setState	@	react.development.js:335
internalSetState	@	downshift.cjs.js:896
setHighlightedIndex	@	downshift.cjs.js:848
(anonymous)	@	ComboBox.js:130
(anonymous)	@	downshift.cjs.js:944
getStateFromUpdate	@	react-dom.development.js:16073
processUpdateQueue	@	react-dom.development.js:16134
updateClassInstance	@	react-dom.development.js:11612
updateClassComponent	@	react-dom.development.js:14490
beginWork	@	react-dom.development.js:15440
performUnitOfWork	@	react-dom.development.js:19108
workLoop	@	react-dom.development.js:19148
renderRoot	@	react-dom.development.js:19231
performWorkOnRoot	@	react-dom.development.js:20138
performWork	@	react-dom.development.js:20050
performSyncWork	@	react-dom.development.js:20024
requestWork	@	react-dom.development.js:19893
scheduleWork	@	react-dom.development.js:19707
enqueueSetState	@	react-dom.development.js:11141
Component.setState	@	react.development.js:335
internalSetState	@	downshift.cjs.js:896
reset	@	downshift.cjs.js:1253
onMouseUp	@	downshift.cjs.js:683
```

**Comment by asudoh** [comment]
Saw that the issue happens in our Storybook (Combobox - Default), too, but it has to be a dev build (and thus _not_ in react.carbondesignsystem.com). As the call stack tells, the issue seems to have something to do with Downshift calling `onInputValueChange` event handler during `setState()`, but we haven't had time to dig in further yet.

**Comment by dakahn** [comment]
@asudoh any further insight into this log message? 📜

**Comment by asudoh** [comment]
@dakahn No, unfortunately. Debugging requires knowledge of how Downshift works internally.
